File: cmd/start.go
package cmd

import "github.com/spf13/cobra"

var host string
var port int
var debug bool

// startCmd represents the start command
var startCmd = &cobra.Command{
	Use:   "start",
	Short: "starts the marathon API server",
	Long: `Starts marathon server with the specified arguments. You can use
environment variables to override configuration keys.`,
	Run: func(cmd *cobra.Command, args []string) {
		// app := api.GetApp(
		// 	host,
		// 	port,
		// 	ConfigFile,
		// 	debug,
		// )
		//
		// app.Start()
	},
}

func init() {
	RootCmd.AddCommand(startCmd)

	startCmd.Flags().StringVarP(&host, "bind", "b", "0.0.0.0", "Host to bind marathon to")
	startCmd.Flags().IntVarP(&port, "port", "p", 8888, "Port to bind marathon to")
	startCmd.Flags().BoolVarP(&debug, "debug", "d", false, "Debug mode")
}

File: src/cmd/start.js
// marathon
// https://github.com/topfreegames/marathon
// Licensed under the MIT license:
// http://www.opensource.org/licenses/mit-license
// Copyright © 2016 Top Free Games <backend@tfgco.com>

import config from 'config'
import MarathonApp from '../api/app'

export default class StartCmd {
  constructor(rootCmd) {
    rootCmd
      .command('start')
      .description('Start API')
      .action(() => {
        const app = new MarathonApp(config)
        app.run()
      })
  }
}
